pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1383,6 +1395,47 @@ func TestBindPodVolumes(t *testing.T) {
 			},
 			shouldFail: true,
 		},
+		"node-affinity-check-fails": {
+			binding:   binding1aBoundSelectedNode,
+			cachedPV:  pvNode1a,
+			cachedPVC: selectedNodePVC,
+			delayFunc: func(t *testing.T, testEnv *testEnv, pod *v1.Pod, pv *v1.PersistentVolume, pvc *v1.PersistentVolumeClaim) {
+				// Update PVC to be fully bound to a PV with a different node
+				newPVC := pvc.DeepCopy()
+				newPVC.ResourceVersion = "100"
+				newPVC.Spec.VolumeName = pvNode2.Name
+				metav1.SetMetaDataAnnotation(&newPVC.ObjectMeta, annBindCompleted, "yes")
+
+				// Update the pod binding cache with the new PVC
+				bindingsCache := testEnv.binder.GetBindingsCache()
+				if bindingsCache == nil {
+					t.Fatalf("Failed to get bindings cache")
+				}
+				bindingsCache.UpdateProvisionedPVCs(pod, pod.Spec.NodeName, []*v1.PersistentVolumeClaim{newPVC})
+
+				// Update PVC cache, fake client doesn't invoke informers
+				internalBinder, ok := testEnv.binder.(*volumeBinder)
+				if !ok {
+					t.Fatalf("Failed to convert to internal binder")
+				}
+
+				pvcCache := internalBinder.pvcCache
+				internalPVCCache, ok := pvcCache.(*pvcAssumeCache)
+				if !ok {
+					t.Fatalf("Failed to convert to internal PVC cache")
+				}
+				internalPVCCache.add(newPVC)
+
+				// Wrong PV isn't added to the PV cache during setup, so add it now
+				pvCache := internalBinder.pvCache
```

Comment:
can this be added in the setup under cachedPV?

Reply:
Yes, but I think I'd need to create a new field. Note that I'm adding to the cache an arbitrary PV bound to a wrong node.

Reply:
That is fine